common/src/main/java/com/wynntils/features/user/ChatCoordinatesFeature.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.managers.Model;
+import com.wynntils.mc.objects.Location;
+import com.wynntils.mc.utils.ComponentUtils;
+import com.wynntils.mc.utils.LocationUtils;
+import com.wynntils.wynn.event.ChatMessageReceivedEvent;
+import com.wynntils.wynn.model.CompassModel;
+import com.wynntils.wynn.utils.WynnUtils;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.stream.Collectors;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.ClickEvent;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.HoverEvent;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.Style;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class ChatCoordinatesFeature extends UserFeature {
+    @Override
+    public List<Class<? extends Model>> getModelDependencies() {
+        return List.of(CompassModel.class);
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public void onChatReceived(ChatMessageReceivedEvent e) {
+        if (!WynnUtils.onWorld()) return;
+
+        Component message = e.getMessage();
+
+        e.setMessage(insertCoordinateComponents(message));
+    }
+
+    public static Component insertCoordinateComponents(Component message) {
+        // no coordinate clickables to insert
+        if (!LocationUtils.strictCoordinateMatcher(ComponentUtils.getCoded(message))
+                .find()) return message;
+
+        List<MutableComponent> components =
+                message.getSiblings().stream().map(Component::copy).collect(Collectors.toList());
+        components.add(0, message.plainCopy().withStyle(message.getStyle()));
+
+        MutableComponent temp = new TextComponent("");
+
+        for (Component comp : components) {
+            Matcher m = LocationUtils.strictCoordinateMatcher((ComponentUtils.getCoded(comp)));
+            if (!m.find()) {
+                Component newComponent = comp.copy();
+                temp.append(newComponent);
+                continue;
+            }
+
+            do {
+                String text = ComponentUtils.getCoded(comp);
```

Comment:
I'm sorry, I don't really have time to fully wrap my head around this. I hope you've thought about how the coordinates might appear in the middle of formatting and that you need to make sure you keep the formatting properly on both sides of the inserted onHover/onClick component.